src/python_testing/TC_OpstateCommon.py

```diff
@@ -1267,67 +1272,83 @@ async def TEST_TC_OPSTATE_BASE_2_6(self, endpoint=1):
             await self.read_and_expect_value(endpoint=endpoint,
                                              attribute=attributes.OperationalState,
                                              expected_value=cluster.Enums.OperationalStateEnum.kRunning)
-            count = sub_handler.attribute_report_counts[attributes.CountdownTime]
-            asserts.assert_greater(count, 0, "Did not receive any reports for CountdownTime")
+            countdownTime = await self.read_expect_success(endpoint=endpoint, attribute=attributes.CountdownTime)
+            if countdownTime is not NullValue:
+                count = sub_handler.attribute_report_counts[attributes.CountdownTime]
+                asserts.assert_greater(count, 0, "Did not receive any reports for CountdownTime")
         else:
             self.skip_step(3)
 
         sub_handler.reset()
         self.step(4)
-        logging.info('Test will now collect data for 30 seconds')
-        time.sleep(30)
-
-        count = sub_handler.attribute_report_counts[attributes.CountdownTime]
-        sub_handler.reset()
-        asserts.assert_less_equal(count, 5, "Received more than 5 reports for CountdownTime")
-        asserts.assert_greater_equal(count, 0, "Did not receive any reports for CountdownTime")
-
-        attr_value = await self.read_expect_success(
-            endpoint=endpoint,
-            attribute=attributes.OperationalState)
-        if attr_value == cluster.Enums.OperationalStateEnum.kRunning:
+        countdownTime = await self.read_expect_success(endpoint=endpoint, attribute=attributes.CountdownTime)
+        if countdownTime is not NullValue:
             self.step(5)
-            wait_count = 0
-            while (attr_value != cluster.Enums.OperationalStateEnum.kStopped) and (wait_count < 20):
-                time.sleep(1)
-                wait_count = wait_count + 1
-                attr_value = await self.read_expect_success(
-                    endpoint=endpoint,
-                    attribute=attributes.OperationalState)
+            logging.info('Test will now collect data for 30 seconds')
+            time.sleep(30)
+
             count = sub_handler.attribute_report_counts[attributes.CountdownTime]
+            sub_handler.reset()
             asserts.assert_less_equal(count, 5, "Received more than 5 reports for CountdownTime")
-            asserts.assert_greater(count, 0, "Did not receive any reports for CountdownTime")
+            asserts.assert_greater_equal(count, 0, "Did not receive any reports for CountdownTime")
         else:
             self.skip_step(5)
 
-        sub_handler.reset()
         self.step(6)
+        countdownTime = await self.read_expect_success(endpoint=endpoint, attribute=attributes.CountdownTime)
+        if countdownTime is not NullValue:
+            attr_value = await self.read_expect_success(
+                endpoint=endpoint,
+                attribute=attributes.OperationalState)
```

Comment:
Suggest pre-reading this to un-nest the if, so that you have:

`if countdownTime is not NullValue and attr_value == cluster.Enums.OperationalStateEnum.kRunning:` in one single if.

Reply:
I had it that way originally, and it makes the code nicer, but it made writing the test steps harder.